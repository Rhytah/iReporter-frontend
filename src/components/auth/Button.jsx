import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ className, children }) => (
  <div className="form-group">
    <button
      className={className}
      type="submit"
    >
      {children}
    </button>
  </div>
);

Button.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
