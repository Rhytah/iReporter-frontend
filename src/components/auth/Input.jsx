import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => {
  const {
    name,
    type,
    value,
    handleChange,
    placeholder,
    onBlur,
    className,
    fieldError,
  } = props;
  return (
    <div className="form-group">
      <input
        className={className}
        name={name}
        type={type}
        value={value}
        onChange={handleChange}
        placeholder={placeholder}
        onBlur={onBlur}
        required
      />
      <div className="invalid-feedback">{fieldError}</div>

    </div>
  );
};
Input.defaultProps = {
  fieldError: '',
};

Input.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
  fieldError: PropTypes.string,
};

export default Input;
