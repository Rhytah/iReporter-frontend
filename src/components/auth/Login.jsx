import React from 'react';

const Login = () => (
  <div>
    <button className="btn btn-primary" type="button" data-toggle="modal" data-target="#modalLogin">
      MODAL LOGIN
    </button>
    <div className="modal fade" role="dialog" tabIndex="-1" id="modalLogin">
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h3 className="modal-title text-primary">
              <i className="fa fa-edit" />
              &nbsp;CREA TU CUENTA
            </h3>
            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">×</span>
            </button>
          </div>
          <div className="modal-body">
            <div className="row">
              <div className="col padMar margenesCajas">
                <p className="text-secondary p-0 m-0 p-2">
                  Para acceder a nuestros productos debes estar inscrito en nuestro sitio web.
                </p>
              </div>
            </div>
            <form method="post" className="p-4">
              <div className="form-group">
                <input
                  className="form-control"
                  type="email"
                  name="email"
                  placeholder="Correo electrónico"
                />
              </div>
              <div className="form-group">
                <input
                  className="form-control"
                  type="password"
                  name="password"
                  placeholder="Contraseña"
                />
              </div>
              <div className="form-group">
                <input
                  className="form-control"
                  type="password"
                  name="password"
                  placeholder="Repetir contraseña"
                />
              </div>
              <div className="form-group">
                <button className="btn btn-primary btn-block" type="submit">
                  ENVIAR
                </button>
              </div>
              <div className="form-check pb-2">
                <input className="form-check-input" type="checkbox" id="formCheck-1" />
                <button type="submit" className="form-check-label text-left" htmlFor="formCheck-1">
                  Recordarme
                </button>
              </div>
              <a href="/" className="forgot">
                ¿Olvidaste tu contraseña?
              </a>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default Login;
