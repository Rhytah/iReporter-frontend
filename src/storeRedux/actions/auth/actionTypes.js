export const RegistrationConstants = {
  REGISTER_REQUEST: 'USERS_REGISTER_REQUEST',
  REGISTER_SUCCESS: 'USERS_REGISTER_SUCCESS',
  REGISTER_FAILURE: 'USERS_REGISTER_FAILURE',
};

export const LoginActionTypes = {
  LOGIN_REQUEST: 'LOGIN_REQUEST',
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_FAILURE: 'LOGIN_FAILURE',
};
