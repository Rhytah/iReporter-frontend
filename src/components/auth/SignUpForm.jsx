/* eslint-disable camelcase */
/* eslint-disable max-len */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as registrationActions from '../../storeRedux/actions/auth/registerActions';
import Input from './Input';
import Button from './Button';
import displayToast from '../../utils/toast';

export class SignUpForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstname: '',
      lastname: '',
      username: '',
      password: '',
      email: '',
      firstnameError: '',
      lastnameError: '',
      usernameError: '',
      emailError: '',
      passwordError: '',
      confirmpassword: '',
      confirmPasswordError: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.submitForm = this.submitForm.bind(this);
    this.validateFormData = this.validateFormData.bind(this);
    this.handleClearForm = this.handleClearForm.bind(this);
  }

  handleChange(event) {
    this.setState(
      {
        [event.target.name]: event.target.value,
      },
      () => {
        this.validateFormData();
      },
    );
  }

  validateFormData() {
    const {
      email, username, password, confirmpassword,
    } = this.state;

    // eslint-disable-next-line no-useless-escape
    const passwordValidation = new RegExp(
      '^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})',
    );
    // eslint-disable-next-line no-useless-escape
    const usernameValidation = new RegExp('^([a-zA-Zd]+[-_])*[a-zA-Zd*]+$');

    this.setState({
      emailError:
        email.length > 3 ? null : 'Email must be longer than 3 characters',
      usernameError:
        username > 3 || usernameValidation.test(username)
          ? null
          : 'Username should be 4 characters and cannot be integers, have white spaces or symbol',
      passwordError: passwordValidation.test(password)
        ? null
        : 'Password should be at least 6 characters with atleast a number, capital and small letter.',
      confirmPasswordError:
        confirmpassword === password ? null : "Passwords don't match",
    });
  }

  submitForm(event) {
    event.persist();
    event.preventDefault();
    const {
      email, username, password, firstname, lastname,
    } = this.state;
    const { actions } = this.props;
    const newUser = {
      email,
      username,
      password,
      firstname,
      lastname,
    };
    actions
      .registerUser(newUser)
      .then((response) => {
        const { history } = this.props;
        Notify.success(response.data.message);
        this.handleClearForm(event);
        history.push('/login');
      })
      .catch(error => Notify.error(error));
  }

  handleClearForm(event) {
    event.preventDefault();
    this.setState({
      firstname: '',
      lastname: '',
      username: '',
      password: '',
      email: '',
      confirmpassword: '',
    });
  }

  render() {
    const {
      firstname,
      lastname,
      username,
      password,
      email,
      firstnameError,
      lastnameError,
      usernameError,
      emailError,
      passwordError,
      confirmpassword,
      confirmPasswordError,
    } = this.state;

    const { user } = this.props;

    return (
      <div className="wrapper fadeInDown">
        <div id="formContent">
          <div className="fadeIn first">
            <img
              src="https://www.tsafindia.org/assets/images/signin.png"
              id="icon"
              alt="User Icon"
            />
          </div>
          <form onSubmit={this.submitForm}>
            <h2 className="text-center">
              <strong>Create </strong>
              an account
            </h2>
            <Input
              name="firstname"
              type="text"
              placeholder="First Name"
              value={firstname}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${firstnameError ? 'is-invalid' : ''}`}
              fieldError={firstnameError}
            />
            <Input
              name="lastname"
              type="text"
              placeholder="Last Name"
              value={lastname}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${lastnameError ? 'is-invalid' : ''}`}
              fieldError={lastnameError}
            />
            <Input
              name="email"
              type="email"
              placeholder="Email"
              value={email}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${emailError ? 'is-invalid' : ''}`}
              fieldError={emailError}
            />
            <Input
              name="username"
              type="text"
              placeholder="Username"
              value={username}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${usernameError ? 'is-invalid' : ''}`}
              fieldError={usernameError}
            />
            <Input
              name="password"
              type="password"
              placeholder="Password"
              value={password}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${passwordError ? 'is-invalid' : ''}`}
              fieldError={passwordError}
            />
            <Input
              name="confirmpassword"
              type="password"
              placeholder="Password (Confirm)"
              value={confirmpassword}
              handleChange={this.handleChange}
              onBlur={this.validateFormData}
              className={`form-control ${
                confirmPasswordError ? 'is-invalid' : ''
              }`}
              fieldError={confirmPasswordError}
            />
            <div className="form-group">
              <Link to="/" className="already">
                Changed your mind? Cancel registration.
              </Link>
            </div>
            {user.registering && <div>Creating Account...</div>}
            <Button className="btn btn-primary">Signup</Button>
            <Link to="/login" className="already">
              You already have an account? Login here.
            </Link>
          </form>
        </div>
      </div>
    );
  }
}

SignUpForm.propTypes = {
  actions: PropTypes.shape({ registerUser: PropTypes.func }).isRequired,
  user: PropTypes.shape({ registerUser: PropTypes.func }).isRequired,
};

export const mapStateToProps = state => ({
  user: state.user,
});

export const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(registrationActions, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUpForm);
