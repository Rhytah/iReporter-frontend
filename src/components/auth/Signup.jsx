/* eslint-disable max-len */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as registrationActions from '../../storeRedux/actions/auth/registerActions';
import Input from './Input';
import Button from './Button';

export class SignUpForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      username: '',
      password: '',
      emailError: '',
      usernameError: '',
      passwordError: '',
      confirmpassword: '',
      confirmPasswordError: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.submitForm = this.submitForm.bind(this);
    this.validateFormData = this.validateFormData.bind(this);
    this.handleClearForm = this.handleClearForm.bind(this);
  }

  handleChange(event) {
    this.setState(
      {
        [event.target.name]: event.target.value,
      },
      () => {
        this.validateFormData();
      },
    );
  }

  validateFormData() {
    const {
      email, username, password, confirmpassword,
    } = this.state;

    // eslint-disable-next-line no-useless-escape
    const passwordValidation = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})');
    // eslint-disable-next-line no-useless-escape
    const usernameValidation = new RegExp('^([a-zA-Zd]+[-_])*[a-zA-Zd*]+$');

    this.setState({
      emailError: email.length > 3 ? null : 'Email must be longer than 3 characters',
      usernameError:
        username > 3 || usernameValidation.test(username)
          ? null
          : 'Username should be 4 characters and cannot be integers, have white spaces or symbol',
      passwordError: passwordValidation.test(password)
        ? null
        : 'Password should be 8 to more characters with atleast a number, capital and small letter.',
      confirmPasswordError: confirmpassword === password ? null : "Passwords don't match",
    });
  }

  submitForm(event) {
    event.preventDefault();
    const { email, username, password } = this.state;
    const { actions } = this.props;
    const newUser = { email, username, password };
    actions.registerUser(newUser);
    this.handleClearForm(event);
  }

  handleClearForm(event) {
    event.preventDefault();
    this.setState({
      email: '',
      username: '',
      password: '',
      confirmpassword: '',
    });
  }

  render() {
    const {
      email,
      username,
      password,
      confirmpassword,
      emailError,
      usernameError,
      passwordError,
      confirmPasswordError,
    } = this.state;

    const { user } = this.props;

    return (
      <div id="page-top">
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal1"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/1-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div className="modal fade" role="dialog" tabIndex="-1" id="modalLogin">
            <div className="modal-dialog" role="document">
              <div className="modal-content">
                <div className="modal-header">
                  <h3 className="modal-title text-primary">
                    <i className="fa fa-edit" />
&nbsp;CREA TU CUENTA
                  </h3>
                  <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                    <span aria-hidden="true">×</span>
                  </button>
                </div>
                <div className="modal-body">
                  <div className="row">
                    <div className="col padMar margenesCajas">
                      <p className="text-secondary p-0 m-0 p-2">
                        Para acceder a nuestros productos debes estar inscrito en nuestro sitio web.
                      </p>
                    </div>
                  </div>
                  <form method="post" className="p-4">
                    <div className="form-group">
                      <input
                        className="form-control"
                        type="email"
                        name="email"
                        placeholder="Correo electrónico"
                      />
                    </div>
                    <div className="form-group">
                      <input
                        className="form-control"
                        type="password"
                        name="password"
                        placeholder="Contraseña"
                      />
                    </div>
                    <div className="form-group">
                      <input
                        className="form-control"
                        type="password"
                        name="password"
                        placeholder="Repetir contraseña"
                      />
                    </div>
                    <div className="form-group">
                      <button className="btn btn-primary btn-block" type="submit">
                        ENVIAR
                      </button>
                    </div>
                    <div className="form-check pb-2">
                      <input className="form-check-input" type="checkbox" id="formCheck-1" />
                      <button type="button" className="form-check-label text-left" htmlFor="formCheck-1">
                        Recordarme
                      </button>
                    </div>
                    <a href="/" className="forgot">
                      ¿Olvidaste tu contraseña?
                    </a>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col">
            <button
              className="btn btn-primary btn-Oscuro"
              type="button"
              data-bs-hover-animate="tada"
              data-toggle="modal"
              data-target="#modal"
            >
              MODAL
            </button>
            <div className="modal fade centro" role="dialog" tabIndex="-1" id="modal">
              <div className="modal-dialog" role="document">
                <div className="modal-content">
                  <div className="modal-header">
                    <h6 className="modal-title txtsGrises">REGISTER </h6>
                    <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">×</span>
                    </button>
                  </div>
                  <div className="modal-body">
                    <h2 className="titulos">Register form: </h2>
                    <p className="text-left margenesEspacio">
                      Write your information to be part of our community and have access to the
                      system.
                      {' '}
                    </p>
                    <div className="col padMar">
                      <input type="text" placeholder="Name" className="margenesEspacio inputLargo" />
                      <input
                        type="text"
                        placeholder="Lastname"
                        className="margenesEspacio inputLargo"
                      />
                      <input type="text" placeholder="Email " className="margenesEspacio inputLargo" />
                      <input
                        type="password"
                        placeholder="Create a password"
                        className="margenesEspacio inputLargo"
                      />
                      <input
                        type="password"
                        placeholder="Confirm your password"
                        className="margenesEspacio inputLargo"
                      />
                      <input
                        type="tel"
                        placeholder="Mobile phone"
                        className="margenesEspacio inputLargo"
                      />
                    </div>
                  </div>
                  <div className="modal-footer">
                    <button className="btn btn-primary btn-Oscuro" type="button">
                      REGISTER
                      {' '}
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal2"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/2-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal3"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/3-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal4"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/4-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal5"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/5-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade portfolio-modal text-center"
          role="dialog"
          tabIndex="-1"
          id="portfolioModal6"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <div className="container">
                  <div className="row">
                    <div className="col-lg-8 mx-auto">
                      <div className="modal-body">
                        <h2 className="text-uppercase">Project Name</h2>
                        <p className="item-intro text-muted">Lorem ipsum dolor sit amet consectetur.</p>
                        <img alt="placeholder-pic" src="/portfolio/6-full.jpg" className="img-fluid d-block mx-auto" />
                        <p>
                          Use this area to describe your project. Lorem ipsum dolor sit amet,
                          consectetur adipisicing elit. Est blanditiis dolorem culpa incidunt minus
                          dignissimos deserunt repellat aperiam quasi sunt officia expedita beatae
                          cupiditate, maiores repudiandae, nostrum, reiciendis facere nemo!
                        </p>
                        <ul className="list-unstyled">
                          <li>Date: January 2017</li>
                          <li>Client: Threads</li>
                          <li>Category: Illustration</li>
                        </ul>
                        <button className="btn btn-primary" type="button" data-dismiss="modal">
                          <i className="fa fa-times" />
                          <span>&nbsp;Close Project</span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SignUpForm.propTypes = {
  actions: PropTypes.shape({ registerUser: PropTypes.func }).isRequired,
  user: PropTypes.shape({ registerUser: PropTypes.func }).isRequired,
};

export const mapStateToProps = state => ({
  user: state.user,
});

export const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(registrationActions, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUpForm);
