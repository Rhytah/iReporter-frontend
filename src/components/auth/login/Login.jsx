import React from 'react';
import PropTypes from 'prop-types';
import '../../../assets/css/Login.css';

const Login = (props) => {
  const {
    handleChange, password, username, handleSubmit,
  } = props;
  console.log(props, 'hjjhj========>');
  return (
    <React.Fragment>
      <div className="wrapper fadeInDown">
        <div id="formContent">
          <div className="fadeIn first">
            <img src="https://www.tsafindia.org/assets/images/signin.png" id="icon" alt="User Icon" />
          </div>

          <form onSubmit={handleSubmit}>
            <input
              type="text"
              className="form-control"
              name="username"
              value={username}
              placeholder="username"
              required="required"
              onChange={handleChange}
            />
            <input
              type="password"
              className="form-control"
              name="password"
              value={password}
              placeholder="Password"
              required="required"
              onChange={handleChange}
            />
            <input type="submit" className="fadeIn fourth" value="Log In" />
          </form>
          <div id="formFooter">
            <a className="underlineHover" href="/">
              Forgot Password?
            </a>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};
Login.propTypes = {
  handleChange: PropTypes.func.isRequired,
  password: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;
