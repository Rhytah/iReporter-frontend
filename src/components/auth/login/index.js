/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Login from './Login';
import * as loginActions from '../../../storeRedux/actions/auth/loginActions';

export class LoginModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      password: '',
    };
  }

  handleChange = (event) => {
    event.preventDefault();
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { actions } = this.props;
    const userData = this.state;

    actions
      .loginInUser(userData)
      .then((data) => {
        Notify.success(data.message);
        this.props.history.push('/');
      })
      .catch((response) => {
        Notify.error(response.error);
      });
  };

  render() {
    const { username, password } = this.state;

    return (
      <Login
        username={username}
        password={password}
        handleChange={this.handleChange}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

LoginModal.propTypes = {
  actions: PropTypes.shape({ loginInUser: PropTypes.func }).isRequired,
};

export const mapStateToProps = state => ({
  loginSate: state.signInReducer,
});

export const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(loginActions, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoginModal);
