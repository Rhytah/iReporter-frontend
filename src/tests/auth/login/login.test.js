import React from 'react';
import { shallow } from 'enzyme';
import { LoginModal } from '../../../components/auth/login';
import Login from '../../../components/auth/login/Login';

function renderLoginModal(args) {
  const defaultProps = {
    actions: {
      loginInUser: jest
        .fn()
        .mockResolvedValue({ data: { message: 'Login was successfull' } }),
    },
    loginState: {},
  };
  const props = { ...defaultProps, ...args };
  return shallow(<LoginModal {...props} />);
}

function renderLogin(args) {
  const defaultProps = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    password: '',
    username: '',
    logInState: false,
  };
  const props = { ...defaultProps, ...args };
  return shallow(<Login {...props} />);
}

describe('Login component ', () => {
  const wrapper = renderLogin();

  it('component should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('LoginModal Component', () => {
  const wrapper = renderLoginModal();
  const wrapperInst = wrapper.instance();

  it('should handle input change', () => {
    const event = {
      target: {
        name: 'username',
        value: 'fred',
      },
      preventDefault: jest.fn(),
    };
    wrapperInst.handleChange(event);
    expect(wrapperInst.state.username).toBe('fred');
  });

  it('should handle login', () => {
    wrapper.setState({
      username: 'fred',
      password: 'password',
    });

    expect(wrapperInst.state.username).toBe('fred');
    expect(wrapperInst.state.password).toBe('password');

    const event = {
      target: {
        type: 'submit',
      },
      preventDefault: jest.fn(),
    };
    wrapperInst.handleSubmit(event);
    expect(wrapperInst.props.actions.loginInUser).toBeCalled();
  });
});
