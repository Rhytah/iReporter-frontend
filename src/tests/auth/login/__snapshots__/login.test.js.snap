// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component  component should match the snapshot 1`] = `
<Fragment>
  <div
    className="container-fluid"
  >
    <div
      className="row no-gutter"
    >
      <div
        className="d-none d-md-flex col-md-4 col-lg-6 bg-image"
      />
      <div
        className="col-md-8 col-lg-6"
      >
        <div
          className="login d-flex align-items-center py-5"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-9 col-lg-8 mx-auto"
              >
                <h3
                  className="login-heading mb-4"
                >
                  Login to Continue
                </h3>
                <form
                  onSubmit={[MockFunction]}
                >
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      name="username"
                      onChange={[MockFunction]}
                      placeholder="username"
                      required="required"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      name="password"
                      onChange={[MockFunction]}
                      placeholder="Password"
                      required="required"
                      type="password"
                      value=""
                    />
                  </div>
                  <button
                    className="btn btn-lg btn-primary btn-block btn-login text-uppercase font-weight-bold mb-2"
                    type="submit"
                  >
                    Log in
                  </button>
                  <div
                    className="text-center"
                  >
                    <a
                      className="small"
                      href="#gggg"
                    >
                      Forgot password?
                    </a>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
